repo/blob/sftp/sftp_storage.go

```diff
@@ -209,7 +209,19 @@ func (s *sftpImpl) PutBlobInPath(ctx context.Context, dirPath, fullPath string,
 			return errors.Wrap(err, "can't close temporary file")
 		}
 
-		err = sftpClientFromConnection(conn).PosixRename(tempFile, fullPath)
```

Comment:
Can we do this as a fallback instead of a regular code path?

Basically we're changing 1 request to the server (posix rename) for 2-3 (stat/remove/rename) which will increase overall latency.  When we discover that server does not support posix rename we can execute slower path and not even attempt posix rename in the future.

BTW. This is also technically incorrect, since if the program were to fail at the worst possible moment, which is after removing the `fullPath`, nobody will execute `rename`, but it probably does not matter that much since Kopia never really rewrites data blobs in place.

Reply:
> Can we do this as a fallback instead of a regular code path?

yup, makes sense to me, I'll need to research if there is a way to check the methods availability first, I'd rather not default to it on every error.

maybe this https://pkg.go.dev/github.com/pkg/sftp#Client.HasExtension would already be enough

> BTW. This is also technically incorrect, since if the program were to fail at the worst possible moment, which is after removing the fullPath, nobody will execute rename, but it probably does not matter that much since Kopia never really rewrites data blobs in place.

true, but there is no alternative standard way to emulate this exact behavior